### PR TITLE
[jk] Add "No tags" filtering option for pipelines

### PR DESCRIPTION
**Pull request by edmondwinston**
# Description
This PR adds a `No tags` filtering option on the `/pipelines` page.

# How Has This Been Tested?
image

filter pipelines with no tags

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

cc: @johnson-mage

### CODE REVIEW

**Comment by johnson-mage** [comment]
@edmondwinston You can install the `ESLint` extension in VSCode to catch minor linting issues like missing semicolons.

File: mage_ai/frontend/pages/pipelines/index.tsx [hunk 1]
```diff
@@ -661,7 +657,15 @@ function PipelineListPage() {
   ]);
 
   const { data: dataTags } = api.tags.list();
-  const tags: TagType[] = useMemo(() => sortByKey(dataTags?.tags || [], ({ uuid }) => uuid), [
+  const tags: TagType[] = useMemo(() => {
+    const t = dataTags?.tags
+    if (t?.length > 0 && t[0].uuid !== 'no_tag') { // <- So as to not to re-add the `no_tag` tag.
```
Comment (johnson-mage) [inline mage_ai/frontend/pages/pipelines/index.tsx:662]: This seems to cause some unintended duplicates for the `no_tag` uuid.
image

**Comment by johnson-mage** [comment]
When including a `no_tag` filter plus some other tags, we would expect the pipelines without tags to be included, as well as the pipelines with the selected tags. It looks like if you include the `no_tag` filter plus some other tag, no pipelines will be displayed in the UI.
image

File: mage_ai/frontend/pages/pipelines/index.tsx (hunk 1 above)
Comment (edmondwinston) [reply mage_ai/frontend/pages/pipelines/index.tsx]: Ah, I was short-sighted - I didn't test the feature in every ways. I also saw you pushing, so it'd be lovely if I could take another issue to make it up.

**Comment by johnson-mage** [comment]
@edmondwinston Sorry to take over the PR. There were several changes that I thought should be made, but it would've taken a lot longer to go through every one, so I thought it'd be easier if you saw the diff after I committed the changes to get an idea of what I was thinking.

**Comment by edmondwinston** [comment]
No worries boss, I truly learnt a lot.

**Review by dy46 — APPROVED** [review]

**Pull request merged**